test/PDPVerifier.t.sol

```diff
@@ -1804,4 +1840,64 @@ contract PDPVerifierMigrateTest is Test {
         vm.expectRevert("InvalidInitialization()");
         UUPSUpgradeable(address(proxy)).upgradeToAndCall(address(newImplementation), migrationCall);
     }
+}
+
+contract PDPVerifierOwnershipListenerTest is Test {
+    PDPVerifier pdpVerifier;
+    TestingRecordKeeperService listener;
+    address public owner;
+    address public nextOwner;
+    address public nonOwner;
+    bytes empty = new bytes(0);
+
+    function setUp() public {
+        PDPVerifier pdpVerifierImpl = new PDPVerifier();
+        bytes memory initializeData = abi.encodeWithSelector(
+            PDPVerifier.initialize.selector,
+            2
+        );
+        MyERC1967Proxy proxy = new MyERC1967Proxy(address(pdpVerifierImpl), initializeData);
+        pdpVerifier = PDPVerifier(address(proxy));
+        listener = new TestingRecordKeeperService();
+        owner = address(this);
+        nextOwner = address(0x1234);
+        nonOwner = address(0xffff);
+    }
+
+    function testOwnerChangedCalledOnOwnershipTransfer() public {
+        // Register listener and create proof set
+        uint256 setId = pdpVerifier.createProofSet{value: PDPFees.sybilFee()}(address(listener), empty);
+        pdpVerifier.proposeProofSetOwner(setId, nextOwner);
+        vm.prank(nextOwner);
+        pdpVerifier.claimProofSetOwnership(setId);
+        // Assert the listener's ownerChanged was called with correct params
+        assertEq(listener.lastOwnerChangedProofSetId(), setId, "Proof set ID mismatch");
+        assertEq(listener.lastOwnerChangedOldOwner(), owner, "Old owner mismatch");
+        assertEq(listener.lastOwnerChangedNewOwner(), nextOwner, "New owner mismatch");
+        assertEq(listener.ownerChangedCallCount(), 1, "ownerChanged should be called once");
+    }
+
+    function testNoListenerNoRevert() public {
+        // Create proof set with no listener
+        uint256 setId = pdpVerifier.createProofSet{value: PDPFees.sybilFee()}(address(0), empty);
+        pdpVerifier.proposeProofSetOwner(setId, nextOwner);
+        vm.prank(nextOwner);
+        // Should not revert even though listenerAddr is zero
```

Comment:
As established in previous comment this behavior should flip after review response.